### PR TITLE
xdg-terminal-exec: Init at 75fa03c9df3c96e16223f23b09f4d47e58a51233

### PR DESCRIPTION
**Pull request by sents**
As mentioned in https://gitlab.gnome.org/GNOME/glib/-/issues/338 this is a temporary solution to enable users to set their prefered terminal emulator for glib.

xdg-terminal-exec requires .desktop like files in a new `$XDG_DATA_DIR/xdg-terminals`,
but adding these files for all of our terminal emulators would be some more work.
As this is only supposed to be a temporary solution I'm not sure it's worth the effort,
although it did take 12 years for a temporary solution to emerge.

In the meantime users can just add their required terminals in `$HOME/.local/share/xdg-terminals`

Homepage: https://github.com/Vladimir-csp/xdg-terminal-exec

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/xdg-terminal-exec/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, lib, fetchFromGitHub, bash }:
+stdenv.mkDerivation rec {
+  pname = "xdg-terminal-exec";
+  version = "0.1.0";
```
Comment (Tungsten842) [inline pkgs/applications/terminal-emulators/xdg-terminal-exec/default.nix:4]:
```suggestion
  version = "unstable-2023-05-12";
```
https://nixos.org/manual/nixpkgs/stable/#sec-package-naming

**Review by Tungsten842 — CHANGES_REQUESTED** [review]

**Review by Tungsten842 — APPROVED** [review]

File: pkgs/applications/terminal-emulators/xdg-terminal-exec/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  preInstall = ''
+  substituteInPlace xdg-terminal-exec \
+    --replace "#!/bin/sh" "${bash}"
+  '';
+  installPhase = ''
+  mkdir -p $out/bin/
+  cp xdg-terminal-exec $out/bin/
+  '';
```
Comment (ncfavier) [inline pkgs/applications/terminal-emulators/xdg-terminal-exec/default.nix:20]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/bin/
    cp xdg-terminal-exec $out/bin/
    runHook postInstall
  '';
```

File: pkgs/applications/terminal-emulators/xdg-terminal-exec/default.nix
Comment (ncfavier) [inline pkgs/applications/terminal-emulators/xdg-terminal-exec/default.nix:1]: Nitpick: not a terminal emulator, I'd put this under `pkgs/tools/misc`.

**Review by ncfavier — COMMENTED** [review]
That shell script looks like an insecure pile of hacks. While I'm not opposed to adding it to nixpkgs, let's not include it in NixOS.

File: pkgs/applications/terminal-emulators/xdg-terminal-exec/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    description = "Proposal for XDG terminal execution utility";
+    homepage = "https://github.com/Vladimir-csp/xdg-terminal-exec";
+    license = licenses.gpl3;
```
Comment (jtojnar) [inline pkgs/applications/terminal-emulators/xdg-terminal-exec/default.nix:25]: `gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

**Comment by aacebedo** [comment]
@sents would it be possible to make the modification requested in your PR? I'll it so solve a problem with thunar and nemo.

**Comment by SuperSandro2000** [comment]
Closing in favor of #273386 because the linked PR is ready to be merged

**Pull request closed without merging**